Teradata/covalent
# feature(data-table): new data-table component

## Description

This branch introduces a new Covalent component `td-data-table`. It was heavily inspired on Angular 1's *md-data-table*.

### What's included?

- Features:
   - [x] Basic table layout
   - [x] Automatic columns when none provided
   - [x] Row selection
   - [x] Multiple or single row selection
   - [x] Sorting
   - [ ] In line edit
- An incomplete example of its usage
- Updated documentation (WIP)

#### Test Steps

- Navigate to http://localhost:4200/#/components/data-table
- Check examples

##### Screenshots or link to CodePen/Plunker/JSfiddle

![datatable](https://cloud.githubusercontent.com/assets/867883/19093065/2fc626ea-8a4f-11e6-9a56-b0bd82b61cf7.gif)
